bedevere/prtype.py

```diff
@@ -0,0 +1,46 @@
+"""Label a pull request based on its type."""
+import enum
+import pathlib
+
+from . import util
+
+LABEL_PREFIX = "type"
+
+
+@enum.unique
+class Category(enum.Enum):
+    """Category of Pull Request."""
+    bugfix = f"{LABEL_PREFIX}-bugfix"
+    documentation = f"{LABEL_PREFIX}-documentation"
+    enhancement = f"{LABEL_PREFIX}-enhancement"
+    performance = f"{LABEL_PREFIX}-performance"
+    security = f"{LABEL_PREFIX}-security"
+    tests = f"{LABEL_PREFIX}-tests"
+
+
+async def add_category(gh, issue, category):
+    """Apply this type label if there aren't any type labels on the PR."""
+    if any(label.startswith("type") for label in util.labels(issue)):
+        return
+    await gh.post(issue["labels_url"], data=[category.value])
+
+
+async def classify_by_files(gh, pull_request, filenames):
+    """Categorize pull request."""
+    issue = await util.issue_for_PR(gh, pull_request)
+    docs = tests = 0
```

Comment:
Why the accumulation? Were you thinking of applying the label based on which type of file it had the most of?

Reply:
I initially was checking for blurbs a different way and was subtracting 1 if an `.rst` file was a blurb.  I left it as a count because I was thinking that the number of labels might grow.  I know it's not good to plan for 'what if' scenarios, but I was thinking about maybe adding a `size` label (xs, s, m, l, xl) would help with reviews.  If someone was looking through the PRs, and there was an xs doc issue, they might be able to review it quickly without having to click on every one to see which ones might be able to be merged quickly.  Especially since labels can be used as filters.

Reply:
Nice idea, but yeah, don't plan for the future. 😄 I keep reading the code and thinking, "where is the numeric value used?" So just drop it for now and if the desire to bring back is there it's easy enough to do. Always try to optimize for readability as that makes it more maintainable (and the future might lead to a different insight on how to measure size which could negate this initial work).